### ISSUE TITLE
Could not find a lambda deployed in this stage/region 

### ISSUE THREAD

**Issue by whatsdis** [serverless/serverless#256]
when I go into a jaws project directory and run jaws dash after creating the greetings hello endpoint successfully, it gives me 

-------------------------------------------
Select Resources To Deploy
-------------------------------------------
   undefined/undefined
     L) l
     E) /greetings/hello - GET
   - - - - -
 >   Deploy Selected -->
AWS: -------------------------------------------
AWS:  Dashboard:  Deploying Lambdas...
AWS: -------------------------------------------
AWS: No selected Lambdas to deploy.
AWS: -------------------------------------------
AWS:  Dashboard:  Deploying Endpoints...
AWS: -------------------------------------------
AWS: Endpoint Deployer:  Deploying endpoint(s) to region "us-west-2"...
AWS: Endpoint Deployer:  "dev - us-west-2": found 1 endpoints to deploy
AWS: Warning: JAWS could not find a deployed Cloudformation template containing
lambda functions.
 [JawsError: Stack with id dev-wl-Exebx-l does not exist]
 name: 'JawsError',
 message: 'Stack with id dev-wl-Exebx-l does not exist',
 messageId: 1 }
AWS: Endpoint Deployer:  "dev - us-west-2": created a new REST API on AWS API G
teway with ID: sfmwggs6st5rk
AWS: Endpoint Deployer:  "dev - us-west-2": found 1 existing resources on API G
teway
AWS: Endpoint Deployer:  "dev - us-west-2 - greetings/hello": created resource:
{ [JawsError: Could not find a lambda deployed in this stage/region with this fu
nction name: l]
  name: 'JawsError',
  message: 'Could not find a lambda deployed in this stage/region with this func
tion name: l',
  messageId: undefined }

The reason I created it in us-west-2 is because I created a stack and deleted it back in us-east-1, after that it's impossible to create any more stacks on us-east-1 as it will say ROLLBACK_COMPLETE, even for brand new stacks.

So I tried to do it all over again on a different region, us-west-2, yet the same results.

I really think JAWS is ground breaking stuff, can't wait to see it up and running.

Thank you!

**Comment by bbk79** [comment]
I *think* I fixed a similar issue by deleting all JAWS created Cloudformation stacks...

**Comment by whatsdis** [comment]
well I went into CF and deleted everything but it somehow keeps coming back.

**Comment by bbk79** [comment]
So it seems to have an issue with the lambda function name, it's supposed to be 'lGreetingsHello' as opposed to 'l'. In my little experience with JAWS, that name comes from the handler path in the relevant awsm.json:

```    "cloudFormation": {
      "Description": "",
      "Handler": "aws_modules/greetings/hello/handler.handler",
      "MemorySize": 1024,
      "Runtime": "nodejs",
      "Timeout": 6
    }```

**Comment by whatsdis** [comment]
any idea why it's `I` and not the full function name?

**Comment by bbk79** [comment]
Whats your handler path in 'aws_modules/greetings/hello/awsm.json'? it should be as above if you just created it with jaws module create... it's an 'L' BTW, prefix for Lambda I'm guessing so all function names have that prefix...

**Comment by whatsdis** [comment]
"Handler": "aws_modules\\greetings\\hello\\handler.handler"

**Comment by bbk79** [comment]
That 'undefined/undefined' is also supposed to be 'greetings/hello'
Hmmm, backslashes... could be a platform specific bug in JAWS. Unfortunately I don't have access to a Windows environment to test this. A wild wild guess would be try changing the backslashes to slash and see if that makes any difference?

**Comment by whatsdis** [comment]
didn't make any difference. same 'undefined/undefined' 

Could this be an issue with running jaws on windows 7?

**Comment by bbk79** [comment]
No ideas TBH, if you look in project issues there seems to be one or two other issues around the handler path in Windows so I'm guessing it might be still a work in progress...

https://github.com/jaws-framework/JAWS/issues/235, https://github.com/jaws-framework/JAWS/issues/217

**Comment by whatsdis** [comment]
I am trying to do this on Ubuntu and I'm still getting this issue https://github.com/jaws-framework/JAWS/issues/258

Not exactly as smooth as advertised but I know this is still work in progress. I'm going to stop for the day as I've got nothing done just wrestling with getting the basics running.

**Comment by bbk79** [comment]
I had a smoother experience over the weekend, starting from scratch got some basic CRUD API up and running and installed and deployed the 'awsm-users' module as well. But yes, this is still very new and a work in progress and my experience wasn't problem free either...

I ended up with a similar CF issue at some point after fiddling with project structure, all I remember on how that was eventually fixed is deleting the CF stacks and perhaps the lambda functions and the API gateway endpoints as well.

**Comment by whatsdis** [comment]
okay well I don't think I can continue because I can't get past the first few steps. I've tried on a different machine, the result is same. 

Since running JAWS, It's now impossible to create any CF on us-east now and no idea how to debug.

**Comment by whatsdis** [comment]
I made sure I used a unique region and unique name. so the project I created on windows 7, if I use that name again, the problem will happen.

greetings/hello API call worked.

My solution: just create a new unique project name and don't use win 7

**Comment by terry-permeance** [comment]
On Windows, you seem to need to fix the backslashes before deploying:

Create the module:

`jaws module create terry hello`

Edit `terry/hello/awsm.json` and change double backslashes to a single forward slash, e.g.:
```
"Handler": "aws_modules/terry/hello/handler.handler", 
```
instead of
```
"Handler": "aws_modules\\terry\\hello\\handler.handler", 
```

Deploy using  ```jaws dash```

**Comment by sixlettervariables** [comment]
Dupe of #217, fixed by #219.

**Comment by whatsdis** [comment]
@terry-permeance I had tried just that much earlier but did not work.
